Add `--` as a seperator of flags and arguments.

jarulsamy opened the pull request:
`--` is common practice to clearly say where the
flags stop and and arguments start.

Fixes #7540 

Given the following example:
```
version: "3"

services:
  test:
    container_name: test
    image: hello-world
```

Current behavior:
```
$ docker-compose run -- test
ERROR: No such service: --
```

Post fix:
```
$ docker-compose run -- test

Hello from docker!
. . .
```
This is just a series of small changes to the `docopt` argparser. 

Signed-off-by: Joshua Arulsamy <joshua.gf.arul@gmail.com>

thaJeztah commented on compose/cli/main.py:
```diff
@@ -257,7 +257,7 @@ def build(self, options):
         contents of its build directory, you can run `docker-compose build` to rebuild it.
 
-        Usage: build [options] [--build-arg key=val...] [SERVICE...]
+        Usage: build [options] [--] [--build-arg key=val...] [SERVICE...]
```
this doesn't look correct, as it would mean `--` can be before `--build-arg` and this means build args would not be processed as option?

jarulsamy replied:
Maybe I misunderstood `--build-arg`, but as part of the POSIX standard, I believe, `--`  is intended to signify no flags are present. 

For example with `rm` it can be used on files with `-` in the name, e.g, `rm -- -foo`

thaJeztah replied:
It means "stop processing" after,
so I could do (silly example);

```
docker-compose build --pull -- --pull
```

to build a service named `--pull` (but also have the `--pull` option set

jarulsamy commented:
I agree, including it in the help / usage may be unnecessary, I will investigate if that is possible. 

Also it appears this is not building, all the tests passed on my machine. I'll write some additional tests to see if I can pinpoint the issue. 

Thanks for the quick feedback!

jarulsamy commented:
I added tests and fixed the order to support the build flags we discussed earlier. I believe it is not possible to hide the `--` option from the help / usage, since `docopt` derives it's CLI functionality from the help menu itself.
